RFC/src/RFC-0140_Syncing_and_seeding.md

```diff
@@ -60,83 +60,93 @@ The aim of this Request for Comment (RFC) is to describe the syncing, seeding an
 
 The syncing process MUST be done in the following steps:
 
-1. Set [SynchronizationState] to `Synchronizing`.
-2. Ask peers for their latest block, so it can get the total Proof-of-Work (PoW). 
-3. Choose the longest chain based on total PoW done on that chain.
-4. Select a connected peer with the longest chain to sync from. This is based on the following criteria:
-   - Does the peer have a high enough [pruning horizon]?
-   - Does the peer allow syncing?
-   - Does the peer have a low latency?
-5. Download all headers from genesis block up onto [current head], and validate the headers as you receive them.
-6. Download Unspent Transaction Output ([UTXO]) set at [pruning horizon]. 
-7. Download all blocks from [pruning horizon] up to [current head]. If the node is doing a 
-complete sync, the [pruning horizon] will be infinite, which means you will download all blocks ever 
-created.
-8. Validate blocks as if they were just mined and then received, in chronological order. 
-
-After this process, the node will be in sync, and will be able to process blocks and transactions normally. 
+1. Set [SynchronizationState] to `header_sync`.
+2. Sync all missing headers from the genesis block to the current chain tip. The initial header sync allows the node to
+   confirm that the syncing peer does indeed have a fully intact chain from which to sync that adheres to this node's
+   consensus rules and has a valid proof-of-work that is higher than any competing chains.
+3. Set [SynchronizationState] to `block_sync`.
+4. Start downloading blocks from sync peer starting with the oldest block in our database. A fresh node will start from 
+   the genesis block.
+5. Download all block up to [current head], validating and adding the blocks to the local chain storage as we go.
+6. Once all blocks have been downloaded up and including the current network tip set the [SynchronizationState] to 
+   `listening`.
+
+After this process, the node will be in sync, and will be able to process blocks and transactions normally as they 
+arrive.  
+
+#### Horizon Sync Process
+
+The horizon sync process MUST be done in the following steps:
+
+1. Set [SynchronizationState] to `header_sync`.
+2. Sync all missing headers from the genesis block to the current chain tip. The initial header sync allows the node to
+   confirm that the syncing peer does indeed have a fully intact chain from which to sync that adheres to this nodes
+   consensus rules and has a valid proof-of-work that is higher than any competing chains.
+3. Set [SynchronizationState] to `horizon_sync`.
+4. Download all kernels from the current network tip back to this node's [pruning horizon].
+5. Validate kernel MMR root against headers.
+6. Download all [utxo]'s from the current network tip back to this node's [pruning horizon].
+7. Validate outputs and [utxo] MMR.
+8. Once all kernels and [utxo]s have been downloaded from the network tip back to this node's [pruning horizon] set 
```

Comment:
Step 8  should be validate balance by comparing the utxo set to the total accumulated balance.
Step 9 should be do block sync to get to the listing state. 

horizon sync, only syncs to the horizon before handing over to normal block sync, it is also the only piece of code currently to check the total chain emmision

Reply:
Ok updated, but just to confirm you are saying that only the horizon sync checks the total chain emission. Why isn't that done with normal block sync?

Reply:
Its done indirectly. 
So normal block propagation only checks that that single block has the correct emission for its height. 
It checks that:
`outputs - inputs = emission of that height. `
So it only checks that the block has increased the total emission by the emission value of that height. If you do this for every block, you can assume that the total is also correct. 

Horizon sync does not check every block, so it needs to check that:
`sum(all_ouputs)  = total accumulated emission at that height.`
It has no blocks to check that the block emission was correct.